src/app/modules/submission/encrypt-submission/encrypt-submission.middleware.ts

```diff
@@ -243,13 +260,50 @@ export const scanAndRetrieveAttachments = async (
       error: triggerLambdaResult.error,
     })
 
-    return res.status(StatusCodes.INTERNAL_SERVER_ERROR).send({
-      message:
-        'Something went wrong while scanning your attachments. Please try again.',
+    const { statusCode, errorMessage } = mapRouteError(
+      triggerLambdaResult.error,
+    )
+    return res.status(statusCode).json({
+      message: errorMessage,
     })
   }
 
-  // TODO(FRM-1302): Step 4: Retrieve attachments from the clean bucket.
+  // Step 4: Retrieve attachments from the clean bucket.
+  const downloadCleanFilesResult = await ResultAsync.combine(
+    triggerLambdaResult.value.map((isAttachment) => {
+      if (isAttachment) {
+        // Retrieve attachment from clean bucket.
+        return downloadCleanFile(
+          isAttachment.cleanFile.cleanFileKey,
+          isAttachment.cleanFile.destinationVersionId,
+        ).map((result) => {
+          // Replace content with file retreived from clean bucket.
+          isAttachment.response.content = result
+          // Replace answer with filename.
+          isAttachment.response.answer = isAttachment.response.filename
```

Comment:
clarification: how does this work, are we replacing anything in `req`?

Reply:
Yup! This is kinda how we receive multipart form data too in `configureMultipartReceiver`. Not sure if there's a better day to do it though? 🤔

Reply:
Done in [f03825e](https://github.com/opengovsg/FormSG/pull/6747/commits/f03825ed92343668572e7295863f26132d4ee342)! Thanks!